test/rule_enforcer.test.ts

```diff
@@ -0,0 +1,177 @@
+import { Clue } from '../src/Clue'
+import { Card } from '../src/Card'
+import { Player } from '../src/Player';
+import { Loiterer } from '../src/Loiterer';
+import { Spymaster } from '../src/Spymaster';
+import { Operative } from '../src/Operative';
+import { GameUtility as gu } from '../src/GameUtility';
+import { RuleEnforcer as re } from '../src/RuleEnforcer';
+import { Color, Team, Turn } from '../src/constants/Constants';
+
+import 'mocha';
+import { expect } from 'chai';
+import WebSocket = require('ws')
+import { mock, instance, when } from 'ts-mockito';
+import { decode } from 'punycode';
+
+describe("Filename: rules_enforcer.test.ts:\n\nRules Enforcer", () => {
+	let mock_ws: WebSocket;
+	let mock_ws_instance: WebSocket;
+
+	before(() => {
+		mock_ws = mock(WebSocket);
+		mock_ws_instance = instance(mock_ws);
+	});
+
+	it("should allow names that have at least one character and only letters, underscores and spaces", () => {
+		expect(re.isValidName('test')).to.be.true;
+		expect(re.isValidName('valid name')).to.be.true;
+		expect(re.isValidName('also_valid_name')).to.be.true;
+		expect(re.isValidName('FINAL also_valid nAme')).to.be.true;
+	});
+	
+	it("should not allow names that anything besides letters, underscores and spaces", () => {
+		expect(re.isValidName('')).to.be.false;
+		expect(re.isValidName('1nvalid name')).to.be.false;
+		expect(re.isValidName('@lso_invalid_name')).to.be.false;
+		expect(re.isValidName('FINAL-also_invalid-nAme')).to.be.false;
+	});
+	
+	it("should allow numbers of guesses between 0 and 9 inclusive", () => {
+		expect(re.isValidNumGuesses(0)).to.be.true;
+		expect(re.isValidNumGuesses(5)).to.be.true;
+		expect(re.isValidNumGuesses(9)).to.be.true;
+	});
+		
+	it("should not allow numbers of guesses outside 0 and 9 inclusive", () => {
+		expect(re.isValidNumGuesses(-1)).to.be.false;
+		expect(re.isValidNumGuesses(10)).to.be.false;
+		expect(re.isValidNumGuesses(999)).to.be.false;
+		expect(re.isValidNumGuesses(-1000)).to.be.false;
+	});
+
+	it("should allow words in dictionary ", () => {
+		expect(re.isValidWord("test")).to.be.true;
+		expect(re.isValidWord("hello")).to.be.true;
+		expect(re.isValidWord("rabbit")).to.be.true;
+		expect(re.isValidWord("coniferous")).to.be.true;
+	});
+
+	it("should not allow words not in dictionary", () => {
+		expect(re.isValidWord("bae")).to.be.false
+		expect(re.isValidWord("porg")).to.be.false;
+		expect(re.isValidWord("asdfasdjflasdjflkasdjflkasdj")).to.be.false;
+		expect(re.isValidWord("supercalifragilistcexpialidocious")).to.be.false;
+	});
+
+	it("should return false for words that are not on the board (regardless of dictionary)", () => {
```

Comment:
The rest of these tests describe the enforcer allowing or not allowing words. This makes more sense than returning true or false, which is specific to `wordOnBoard`. In this case, I'd probably rephrase this to say `wordOnBoard should return false...` so that it is clear the general enforcer should not be disallowing words because they aren't on the board.